client/src/components/aggregations/AggregationWidgets.js

```diff
@@ -0,0 +1,296 @@
+import "@fullcalendar/core/main.css"
+import dayGridPlugin from "@fullcalendar/daygrid"
+import "@fullcalendar/daygrid/main.css"
+import FullCalendar from "@fullcalendar/react"
+import BarChart from "components/BarChart"
+import LikertScale from "components/graphs/LikertScale"
+import Pie from "components/graphs/Pie"
+import Leaflet from "components/Leaflet"
+import _escape from "lodash/escape"
+import _isEmpty from "lodash/isEmpty"
+import _uniqueId from "lodash/uniqueId"
+import { Location } from "models"
+import { AssessmentPeriodPropType, PeriodPropType } from "periodUtils"
+import PropTypes from "prop-types"
+import React, { useMemo, useRef, useState } from "react"
+import { Button, Collapse, Table } from "react-bootstrap"
+import utils from "utils"
+
+const DATE_FORMAT = "YYYY-MM-DD"
+
+const aggregationWidgetPropTypes = {
+  values: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.arrayOf(
+      PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+        PropTypes.array,
+        PropTypes.object
+      ])
+    )
+  ]),
+  fieldConfig: PropTypes.object,
+  fieldName: PropTypes.string,
+  vertical: PropTypes.bool,
+  period: PropTypes.oneOfType([AssessmentPeriodPropType, PeriodPropType]),
+  whenUnspecified: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
+}
+const aggregationWidgetDefaultProps = {
+  whenUnspecified: null
```

Comment:
What's the difference between `undefined` and the default of `null` here?

Reply:
I use the whenUndefined to return it when there is statistics data but not for a given field. In general I want to return null in that case (that's why the default of null).